lib/dotcom/trip_plan/itinerary_groups.ex

```diff
@@ -8,48 +8,34 @@ defmodule Dotcom.TripPlan.ItineraryGroups do
 
   import DotcomWeb.TripPlanView, only: [get_one_way_total_by_type: 2]
 
-  alias Dotcom.TripPlan.{Itinerary, Leg, PersonalDetail, TransitDetail}
+  alias Dotcom.TripPlan.{Itinerary, ItineraryGroup, Leg, PersonalDetail, TransitDetail}
   alias OpenTripPlannerClient.ItineraryTag
 
   @short_walk_threshold_minutes 5
-  @max_per_group 5
-
-  @type summarized_leg :: %{
-          routes: [Routes.Route.t()],
-          walk_minutes: non_neg_integer()
-        }
-  @type summary :: %{
-          accessible?: boolean() | nil,
-          duration: non_neg_integer(),
-          first_start: DateTime.t(),
-          first_stop: DateTime.t(),
-          next_starts: [DateTime.t()],
-          summarized_legs: [summarized_leg()],
-          tag: String.t(),
-          total_cost: non_neg_integer(),
-          walk_distance: float()
-        }
-
-  @spec from_itineraries([Itinerary.t()]) :: [
-          %{itineraries: [Itinerary.t()], summary: summary()}
-        ]
-  def from_itineraries(itineraries) do
+  @max_per_group 4
+
+  @doc """
+  From a large list of itineraries, collect them into 5 groups of at most
+  #{@max_per_group} itineraries each, sorting the groups in favor of tagged
+  groups first
+  """
+  @spec from_itineraries([Itinerary.t()], boolean()) :: [ItineraryGroup.t()]
+  def from_itineraries(itineraries, take_from_end? \\ false) do
```

Comment:
Nit: The interface for this function would be clearer if `take_from_end?` were passed in as a keyword arg or an options map, rather than a plain old boolean.

That's shown off pretty clearly by how this function is invoked in the test 👇 
```elixir
sorted_itineraries
|> ItineraryGroups.from_itineraries(true)
```

What `true` means in this context is pretty tough to tease out. It would be a lot clearer if the function were invoked like

```elixir
sorted_itineraries
|> ItineraryGroups.from_itineraries(:take_from_end)
```

Or

```elixir
sorted_itineraries
|> ItineraryGroups.from_itineraries(take_from_end: true)
```

Or

```elixir
sorted_itineraries
|> ItineraryGroups.from_itineraries(
     %{take_from_end: true}
   )
```

Reply:
Ended up using keyword here, mostly because we often use that approach elsewhere in Dotcom, thanks!